src/app-init/MfeDownloadManager.js

```diff
@@ -3,24 +3,43 @@ import { useDispatch, useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
 import { getUsername } from '@entando/apimanager';
 
-import { fetchMfeConfigList } from 'state/mfe/actions';
+import { fetchInitialMfeConfigList } from 'state/mfe/actions';
+import StartupWaitScreen from 'app-init/StartupWaitScreen';
 import RowSpinner from 'ui/pages/common/RowSpinner';
 
 export default function MfeDownloadManager(props) {
   const { children } = props;
   const dispatch = useDispatch();
   const [loading, setLoading] = useState(true);
+  const [isPolling, setIsPolling] = useState(false);
   const currentUserName = useSelector(getUsername);
 
   useEffect(() => {
     // wait until apiManager is not initialised and only after that fetch the mfe config list
     if (currentUserName) {
-      dispatch(fetchMfeConfigList()).then(() => setLoading(false));
+      const fetchConfig = () => {
+        dispatch(fetchInitialMfeConfigList()).then(() => {
+          setLoading(false);
+          setIsPolling(false);
+        }).catch(() => {
+          // if fails, start polling until it succeed
+          setLoading(false);
+          setIsPolling(true);
+          setTimeout(fetchConfig, 10000);
+        });
+      };
```

Comment:
you need to [clear](https://developer.mozilla.org/en-US/docs/Web/API/clearTimeout) `setTimeout` after the execution, this can cause bugs that are uneasy to find out.

Reply:
humm I never heard about clearing a setTimeout... I did small research and looks like it is not really necessary:
https://stackoverflow.com/questions/7391567/when-using-settimeout-do-you-have-to-cleartimeout#:~:text=You%20don't%20actually%20need,because%20setInterval%20usually%20runs%20indefinitely.

Reply:
since we are not storing the timeout instance, I guess the clearing doesn't need to happen. If you look at the docs example, it is storing the setTimeout instance on an object attribute, which is different from what we are doing here.